cmake/sysbuild/image_signing_extra.cmake

```diff
@@ -0,0 +1,132 @@
+if(num_binaries GREATER 0)
+  math(EXPR last_index "${num_binaries} - 1")
+
+  set(signed_paths "")
+  set(signed_names "")
+
+  foreach(index RANGE ${last_index})
+    math(EXPR slot_number "${index} + 1")
+
+    if(NOT DEFINED NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER OR NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER EQUAL -1)
+      message(FATAL_ERROR "Extra image slot ${slot_number} not configured. Set SB_CONFIG_MCUBOOT_EXTRA_IMAGES to at least ${num_binaries} in sysbuild.")
+    endif()
+
+    set(image_id ${NCS_MCUBOOT_EXTRA_${slot_number}_IMAGE_NUMBER})
+
+    list(GET extra_paths ${index} input_bin)
+    list(GET extra_names ${index} image_name)
+    list(GET extra_versions ${index} version)
+
+    if(NOT image_name OR image_name STREQUAL "")
+      get_filename_component(image_name "${input_bin}" NAME_WE)
+    endif()
+
+    set(output_bin "${CMAKE_BINARY_DIR}/${image_name}.signed.bin")
+    set(output_hex "${CMAKE_BINARY_DIR}/${image_name}.signed.hex")
+
+    set(imgtool_sign 
+      ${PYTHON_EXECUTABLE} ${IMGTOOL} sign
+      --version ${version}
+      --align 16
+      --slot-size $<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_PRIMARY_${image_id}_SIZE>
+      --pad-header
+      --header-size ${SB_CONFIG_PM_MCUBOOT_PAD}
+    )
+
+    set(imgtool_extra "")
+    if(NOT "${keyfile}" STREQUAL "")
+      list(APPEND imgtool_extra -k "${keyfile}")
+    endif()
+
+    # Add SHA algorithm based on MCUboot configuration (same as image_signing.cmake)
+    sysbuild_get(CONFIG_MCUBOOT_BOOTLOADER_USES_SHA512 IMAGE mcuboot VAR CONFIG_MCUBOOT_BOOTLOADER_USES_SHA512 KCONFIG)
+    if(CONFIG_MCUBOOT_BOOTLOADER_USES_SHA512)
+      list(APPEND imgtool_extra --sha 512)
+    endif()
+
+    sysbuild_get(CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION IMAGE ${DEFAULT_IMAGE} VAR CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION KCONFIG)
+    if(CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION)
+      sysbuild_get(CONFIG_MCUBOOT_HW_DOWNGRADE_PREVENTION_COUNTER_VALUE IMAGE ${DEFAULT_IMAGE} VAR CONFIG_MCUBOOT_HW_DOWNGRADE_PREVENTION_COUNTER_VALUE KCONFIG)
+      list(APPEND imgtool_extra --security-counter ${CONFIG_MCUBOOT_HW_DOWNGRADE_PREVENTION_COUNTER_VALUE})
+    endif()
+
+    add_custom_command(
+      OUTPUT ${output_bin}
+      COMMAND ${imgtool_sign} ${imgtool_extra} ${input_bin} ${output_bin}
+      DEPENDS ${input_bin}
+    )
+
+    if (SB_CONFIG_SOC_SERIES_NRF52X)
+      set(ext_flash_offset 0x12000000)
+    else()
+      set(ext_flash_offset 0x10000000)
+    endif()
+
+    add_custom_command(
+      OUTPUT ${output_hex}
+      COMMAND ${PYTHON_EXECUTABLE}
+        -c "import sys; import intelhex; intelhex.bin2hex(sys.argv[1], sys.argv[2], int(sys.argv[3], 16) + int(sys.argv[4], 16)) "
+        ${output_bin}
+        ${output_hex}
+        $<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_PRIMARY_${image_id}_ADDRESS>
+        $<IF:$<STREQUAL:$<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_PRIMARY_${image_id}_REGION>,external_flash>,${ext_flash_offset},0>
```

Comment:
doesn't this also have an external flash address? Or must that still be done manually?

Reply:
There was no such configuration previously. I added a generic sysbuild option to determine the address used by flashing tools for external flash programming.